test/index.test.ts

```diff
@@ -771,6 +771,48 @@ describe('Unit | Utility | changeset', () => {
     expect(dummyModel.name.short).toBe('foo');
   });
 
+  it('#set nested objects can contain arrays', () => {
+    expect.assertions(5);
+
+    dummyModel.contact = {
+      emails: ['bob@email.com', 'the_bob@email.com']
+    };
+
+    expect(get(dummyModel, 'contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    const dummyChangeset = Changeset(dummyModel, lookupValidator(dummyValidations));
+
+    expect(dummyChangeset.get('contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    dummyChangeset.set('contact.emails.0', 'fred@email.com');
+
+    expect(dummyChangeset.get('contact.emails.0')).toEqual('fred@email.com');
+
+    dummyChangeset.rollback();
+    expect(dummyChangeset.get('contact.emails')).toEqual(['bob@email.com', 'the_bob@email.com']);
+
+    dummyChangeset.set('contact.emails.0', 'fred@email.com');
+    dummyChangeset.set('contact.emails.1', 'the_fred@email.com');
```

Comment:
Could we have an assertion where `get/set` is out of range?

Reply:
on the lower bound, out of range is only negative, I can add tests for that, but on the upper range, there is no out of bounds, because this is how you'd add stuff.
![image](https://user-images.githubusercontent.com/199018/108102768-fda6d680-7056-11eb-8603-403db96e0346.png)

Reply:
Oh right.  But do you think it is worthwhile adding a test for this custom implementation when contact.emails is only len 2 but we do the following?

`dummyChangeset.set('contact.emails.10', 'the_fred@email.com');`

Reply:
yup! I'm working on adding more tests now (as soon as I figure out how to make `changeset.get('some array').toEqual(some array)`

Reply:
TIL 
![image](https://user-images.githubusercontent.com/199018/108126984-8d10b180-7078-11eb-977a-334655b3dd61.png)

I think this should probs be a hard error? even though behavior is kinda undefined in native arrays?